### PR TITLE
Implement UpdateNamedEntity in flyte-cli

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
This change implements setting/updating the description of a named entity.  For background, a NamedEntity includes a task, workflow or launch plan for a specific {Project, Domain, Name} combination across versions. NamedEntities for all resource types have mutable descriptions. Furthermore, workflows also have a state attribute which enables their visibility in the console.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/235

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from __future__ import absolute_import
 import flytekit.plugins
 
-__version__ = '0.7.0b1'
+__version__ = '0.7.1'
```
Comment (katrogan) [inline flytekit/__init__.py:4]: Please let me know which version I should be bumping this to

Comment (wild-endeavor) [reply flytekit/__init__.py:4]: what's the timeline for this?  how soon do you need it?

**Comment by katrogan** [comment]
cc @joannepolsky
